File: backend-dotnet/SahayaSetu/Controllers/PaymentController.cs
using Microsoft.AspNetCore.Mvc;
using System.Text.Json;

namespace SahayaSetu.Controllers
{
    [Route("api/[controller]")]
    [ApiController]
    public class PaymentController : ControllerBase
    {
        private readonly IConfiguration _configuration;

        public PaymentController(IConfiguration configuration)
        {
            _configuration = configuration;
        }

        [HttpPost("create-order")]
        public IActionResult CreateOrder([FromBody] JsonElement data)
        {
            var amount = data.GetProperty("amount").GetInt32();
            
            // Create a simple order response (matching Java structure)
            var order = new
            {
                id = $"order_{Guid.NewGuid()}",
                amount = amount * 100, // Convert to paise
                currency = "INR",
                receipt = "donation_receipt_001",
                status = "created"
            };

            return Ok(JsonSerializer.Serialize(order));
        }

        [HttpPost("success")]
        public async Task<IActionResult> UpdateDonation([FromBody] JsonElement data)
        {
            var requestId = data.GetProperty("requestId").GetInt32();
            var donorId = data.GetProperty("donorId").GetInt32();
            var amount = data.GetProperty("amount").GetDecimal();
            var paymentId = data.GetProperty("paymentId").GetString();

            // This would integrate with your donation service
            // For now, return success to match Java behavior
            return Ok("Donation recorded successfully");
        }
    }
}

File: backend-java/sahayaSetu/src/main/java/com/sahayaSetu/controllers/PaymentController.java
package com.sahayaSetu.controllers;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;

import lombok.RequiredArgsConstructor;
import org.json.JSONObject;
import org.springframework.beans.factory.annotation.Value;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.PostMapping;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.razorpay.Order;
import com.razorpay.RazorpayClient;
import com.razorpay.RazorpayException;
import com.sahayaSetu.entities.Donation;
import com.sahayaSetu.entities.Donor;
import com.sahayaSetu.entities.FundraiserRequest;
import com.sahayaSetu.entities.Request;
import com.sahayaSetu.entities.enums.PaymentMode;
import com.sahayaSetu.repositories.DonationRepository;
import com.sahayaSetu.repositories.DonorRepository;
import com.sahayaSetu.repositories.RequestRepository;

@RestController
@RequestMapping("/api/payment")
@RequiredArgsConstructor
public class PaymentController {

    @Value("${razorpay.key.id}")
    private String keyId;

    @Value("${razorpay.key.secret}")
    private String keySecret;

    private final DonationRepository donationRepository;
    private final RequestRepository requestRepository;
    private final DonorRepository donorRepository;

    @PostMapping("/create-order")
    public ResponseEntity<?> createOrder(@RequestBody Map<String, Object> data)
            throws RazorpayException {
        
        int amount = Integer.parseInt(data.get("amount").toString());
        RazorpayClient client = new RazorpayClient(keyId, keySecret);
        
        JSONObject options = new JSONObject();
        options.put("amount", amount * 100);
        options.put("currency", "INR");
        options.put("receipt", "donation_receipt_001");
        
        Order order = client.orders.create(options);
        return ResponseEntity.ok(order.toString());
    }

    @PostMapping("/success")
    public ResponseEntity<?> updateDonation(@RequestBody Map<String, Object> data) {
        try {
            Long requestId = Long.parseLong(data.get("requestId").toString());
            Long donorId = Long.parseLong(data.get("donorId").toString());
            BigDecimal amount = new BigDecimal(data.get("amount").toString());
            String paymentId = data.get("paymentId").toString();

            Request req = requestRepository.findById(requestId)
                    .orElseThrow(() -> new RuntimeException("Request not found"));

            if (!(req instanceof FundraiserRequest)) {
                throw new RuntimeException("Not a fundraiser request");
            }

            FundraiserRequest fundraiser = (FundraiserRequest) req;
            Donor donor = donorRepository.findById(donorId)
                    .orElseThrow(() -> new RuntimeException("Donor not found"));

            Donation donation = new Donation();
            donation.setAmount(amount);
            donation.setDonatedAt(LocalDateTime.now());
            donation.setDonor(donor);
            donation.setFundraiserRequest(fundraiser);
            donation.setPaymentMode(PaymentMode.ONLINE);
            donation.setPaymentReference(paymentId);
            donationRepository.save(donation);

            fundraiser.setCollectedAmount(fundraiser.getCollectedAmount().add(amount));
            requestRepository.save(fundraiser);

            return ResponseEntity.ok("Donation recorded successfully");
        } catch (Exception e) {
            return ResponseEntity.status(500).body("Error recording donation: " + e.getMessage());
        }
    }
}
